nf_deployments/v1alpha1/nf_deployment_types.go

```diff
@@ -62,14 +62,14 @@ type NFDeploymentSpec struct {
 
 type ObjectReference struct {
 	// APIVersion of the target resources
-	APIVersion string `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`
+	APIVersion string `yaml:"apiVersion" json:"apiVersion"`
 
 	// Kind of the target resources
-	Kind string `yaml:"kind,omitempty" json:"kind,omitempty"`
+	Kind string `yaml:"kind" json:"kind"`
 
 	// Name of the target resource
 	// +optional
-	Name *string `yaml:"name" json:"name"`
+	Name *string `yaml:"name,omitempty" json:"name,omitempty"`
```

Comment:
If the name is optional, does it mean that there could be multiple reference of a particular kind?

Reply:
it is optional as it allows us to get all the CR(s) of the GV.